updater/updateInfo.go

```
@@ -0,0 +1,89 @@
+package updater
+
+import (
+	"github.com/kinvolk/go-omaha/omaha"
+)
+
+// UpdateInfo is a wrapper around CheckForUpdates response
+// and it provides helper functions.
+type UpdateInfo struct {
+	HasUpdate bool
+	app       *omaha.AppResponse
+	omahaResp *omaha.Response
+}
+
+// NewUpdateInfo creates and returns *UpdateInfo from omaha.Response and appID.
+func NewUpdateInfo(resp *omaha.Response, appID string) *UpdateInfo {
+	if resp == nil {
+		return nil
+	}
+	app := resp.GetApp(appID)
+	if app == nil {
+		return nil
+	}
+	return &UpdateInfo{
+		HasUpdate: app != nil && app.Status == omaha.AppOK && app.UpdateCheck.Status == "ok",
+		app:       app,
+		omahaResp: resp,
+	}
+}
+
+// GetVersion retuns the manifest version of the UpdateInfo.
+func (u *UpdateInfo) GetVersion() string {
+	if u.app == nil || u.app.UpdateCheck == nil || u.app.UpdateCheck.Manifest == nil {
```

If `UpdateInfo` doesn't use a pointer for app, we can slightly simplify this check. The other two, as they are in the omaha pkg, will remain.

This applies to other functions. AFAICS, we don't use the `nil` to easily know it was not initialized, so I guess it should simplify the code. But not completely convinced :)

---

I usually don't like to copy data (probably overcare from my C++ days), but in any case this is internal logic and I don't think it's so important as to make the code unreadable. So let's just keep it and try to move on.

---

Ok, if you prefer to keep it... Note the pointer issue is not limited to this, and there are several doubtious examples in the interfaces too (search for pointers in this page, after loading all the comments). And I didn't highlighted all the examples of pointers on the interfaces, that easily make the code buggy (not handling nil) or more complex (need to return an error now, in case it is nil, etc.).

IMHO, it is simpler to just clean it up of the uneeded pointers. But as you prefer, really :)